examples/btc-to-evm-fungible-transfer/src/blockstream-api.ts

```diff
@@ -0,0 +1,172 @@
+import { TypeOfAddress } from "@buildwithsygma/bitcoin";
```

Comment:
file can be renamed to `blockstreamApi.ts`


Ok so I have a question about this file, I see that this is not part of bitcoin sdk package but used in the example so my question is, if someone is going to use our SDK in future for bitcoin transfers, does this mean they would need utils like this on their own? Why shouldn't we provide these out of the box

furthermore, is there a standard followed by these bitcoin api providers that is generalized and can be added to sdk

Reply:
by doing this we will actually enhance developer experience of using the SDK

you won't have to `processUtxos` just like in example because sdk would already do that for you

c.c @MakMuftic @mpetrunic

Reply:
> file can be renamed to `blockstreamApi.ts`
> 
> Ok so I have a question about this file, I see that this is not part of bitcoin sdk package but used in the example so my question is, if someone is going to use our SDK in future for bitcoin transfers, does this mean they would need utils like this on their own? Why shouldn't we provide these out of the box
> 
> furthermore, is there a standard followed by these bitcoin api providers that is generalized and can be added to sdk

IMHO dev integrating the SDK should handle and take care about getting utxo info or even calculating in their way the size to aproximate the fee to pay. The SDK should care about constructing the transaction with the outputs to use to relay the assets and sign it outside. 

AFAIK there is no standar way of calculating the size of the TX which relies on the amount of inputs + the fee rate that you want to use, or at least this is no so straightforward to do this like in EVM. This utils that we have here for the example are just there to give the dev and idea of the things to consider before using our SDK and also what to pass as parameter for constructing the transfer

Reply:
These are excellent points @saadjhk. 
My thoughts, but let me know what you think: 
We leave the SDK agnostic to this, so developers integrating with the SDK can use any other logic or "provider" if they want. However, we offer these as BTC utility functions that use Blockstream under the hood.

Reply:
I didn't really explain it properly tbh :sweat_smile: 

So, in my opinion, it makes sense to pull these functions into SDK but leave current deposit transaction creation logic agnostic to these (as they already are). External developers then can:
- use our util functions that are based on Blockstream API (easy integration) 
- write their own util functions (maybe they also already have this functionality implemented on their side for regular BTC transactions)


wdyt @saadjhk @wainola

Reply:
@MakMuftic so if I understand this correctly, we would add this functionality into the SDK -utils package perhaps- so it is optional for the dev to use this, wouldn't we? 

My concerns here are:
* the blockstream functions only work if you pass the blockstream API URL to the function, since the endpoints being query here are particular to that API.
* I wouldn't include the function that calculates the size or I will limit that function to be use only for testnet, since that is one way to get the size

For the first concern, since the `esplora` API endpoints are public and open [here](https://github.com/Blockstream/esplora/blob/master/API.md), we can just hardcode this url into the utils package, then leverage this internally, instead of passing this as parameter to the function

Reply:
@wainola Yes, we added an optional utility that you can use if you want to use Blockstream API out of the box. The important thing to highlight is that we don't want to force the developers to use this, so the rest of the SDK can not depend on these util functions.

- yep, we hardcode the URL for API endpoints
- what exactly do you mean by this, would it not work on BTC mainnet?

Reply:
```
what exactly do you mean by this, would it not work on BTC mainnet?
```

Here I mean that this way of calculating the size is not standard. I got it after reading some comments on a issue on btc.js lib. Hence I would just leave that function at the example level and avoid adding this into the utils package. My reasoning here is that perhaps the devs using the SDK either have a better and more accurate approach to this, or the calculate size adding the bytes of the raw tx string.

Reply:
added